File: backend/e-learn/src/main/java/com/example/e_learn/model/Document.java
package com.example.e_learn.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

@Entity
public class Document {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    
    @Column(length=2048)
    private String fileUrl; // URL to PDF file in MinIO
    
    private String description;
    
    private int orderIndex; // Order of document within the lesson
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "lesson_id", nullable = false)
    @JsonIgnore // prevent infinite recursion when serializing
    private Lesson lesson;

    // Getters and setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    
    public String getFileUrl() { return fileUrl; }
    public void setFileUrl(String fileUrl) { this.fileUrl = fileUrl; }
    
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    
    public int getOrderIndex() { return orderIndex; }
    public void setOrderIndex(int orderIndex) { this.orderIndex = orderIndex; }
    
    public Lesson getLesson() { return lesson; }
    public void setLesson(Lesson lesson) { this.lesson = lesson; }
}

File: frontend/elearn_front/lib/models/document.dart
class Document {
  final int id;
  final String title;
  final String? fileUrl;
  final String? description;
  final int orderIndex;

  Document({
    required this.id,
    required this.title,
    this.fileUrl,
    this.description,
    required this.orderIndex,
  });

  factory Document.fromJson(Map<String, dynamic> json) {
    return Document(
      id: json['id'] ?? 0,
      title: json['title'] ?? '',
      fileUrl: json['fileUrl'],
      description: json['description'],
      orderIndex: json['orderIndex'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'fileUrl': fileUrl,
      'description': description,
      'orderIndex': orderIndex,
    };
  }
}




